Pearson-Advance/frontend-app-institution-portal
# initial config, is missing fix the update on reload

# Ticket
https://agile-jira.pearson.com/browse/PADV-1414

# Description
In this PR is fixed the error at the moment to open a new tab, the reference of the institution is lost.

## Changes made
- Refactor in the selector, is been moved to a different file, is included the respective test.
- Added new custom hook to handle the query param using the redux context.
- Refactor in ALL application links to reflect the institution selected. 
- Added new dep to help at the moment to test a custom hook.
- Refactor to the function resposable to format the select option along with it test.

## Screenshot
![image](https://github.com/user-attachments/assets/d6bf4850-8eb8-474c-b8b7-2aebb8e35320)

## How to test
- Make sure to have the devstack runnint
- Instal deps `@testing-library/react-hooks` has been added.
```bash
  npm i 
```
- Go to http://localhost:1980/
- The URL must change automatically to reflect the current institution id.
 -- If you have multiple institutions, change it using the selector, then copy the new url, you should be able to recover the institution selected in the previous tab, along with it information.
